src/main.cc

```diff
@@ -1,6 +1,48 @@
-#include "aqdata/aq_base.h"
+#ifdef TEST
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <unistd.h>
+#include <getopt.h>
+#include <deal.II/base/mpi.h>
 
-int main ()
-{
+#include "test_helpers/bart_test_helper.h"
+
+#endif
+
+//#include "aqdata/aq_base.h"
+
+#ifdef TEST
+int main(int argc, char* argv[]) {
+  // Parse optional arguments
+  ::testing::InitGoogleMock(&argc, argv);
+  
+  int option_index = 0;
+  
+  const struct option longopts[] =
+  {
+    {"report", no_argument, 0, 'r'}
+  };
+  
+  int c;
+  while ((c = getopt_long (argc, argv, "rd:", longopts, &option_index)) != -1)
+    switch(c) {
+      case 'r':
+        btest::GlobalBartTestHelper().SetReport(true);
+        break;
+      case 'd':
+        btest::GlobalBartTestHelper().SetGoldFilesDirectory(optarg);
+        break;
+    }
```

Comment:
Add a default case pls. It sometimes helps prevent undetected errors. If those two cases are the only cases that you can think of, put a `default` with merely `break` in it.